virtuaaliviivakoodi/deconstructors/reference.py

```diff
@@ -0,0 +1,21 @@
+from virtuaaliviivakoodi.constants import SymbolVersion
+from virtuaaliviivakoodi.exceptions import InvalidVersionException
+
+
+def deconstruct_reference_and_version(virtuaaliviivakoodi: str) -> tuple[str, str]:
+    """Returns the reference number in a human-readable format and the version of the virtuaaliviivakoodi"""
+
+    version = virtuaaliviivakoodi[0]
+
+    if version not in [version.value for version in SymbolVersion]:
+        raise InvalidVersionException("Invalid version number. Must be 4 or 5.")
+
+    if version == SymbolVersion.VERSION_4.value:
+        reference: str = virtuaaliviivakoodi[28:48].lstrip("0")
+
+    if version == SymbolVersion.VERSION_5.value:
+        reference: str = "RF" + virtuaaliviivakoodi[25:27] + virtuaaliviivakoodi[28:48].lstrip("0")
```

Comment:
Is this correct 🤔 Slices are 25:27 and 28:48 so character at position 27 is never used?

Reply:
Good catch! The tests missed this due to all the references having the 0 padding.